x-pack/plugins/synthetics/server/alert_rules/common.ts

```diff
@@ -94,34 +168,93 @@ export const setRecoveredAlertsContext = ({
 
-    const state = alert.getState() as SyntheticsCommonState;
+    const state = alert.getState() as SyntheticsCommonState & SyntheticsMonitorStatusAlertState;
 
     let recoveryReason = '';
+    let recoveryStatus = i18n.translate(
+      'xpack.synthetics.alerts.monitorStatus.defaultRecovery.status',
+      {
+        defaultMessage: `has recovered`,
+      }
+    );
     let isUp = false;
+    let linkMessage = '';
 
     if (state?.idWithLocation && staleDownConfigs[state.idWithLocation]) {
       const { idWithLocation } = state;
       const downConfig = staleDownConfigs[idWithLocation];
       if (downConfig.isDeleted) {
-        recoveryReason = i18n.translate('xpack.synthetics.alerts.monitorStatus.deleteMonitor', {
-          defaultMessage: `Monitor has been deleted`,
-        });
+        recoveryStatus = i18n.translate(
+          'xpack.synthetics.alerts.monitorStatus.deleteMonitor.status',
+          {
+            defaultMessage: `has been deleted`,
+          }
+        );
+        recoveryReason = i18n.translate(
+          'xpack.synthetics.alerts.monitorStatus.deleteMonitor.reason',
+          {
+            defaultMessage: `The monitor has been deleted`,
+          }
+        );
       } else if (downConfig.isLocationRemoved) {
-        recoveryReason = i18n.translate('xpack.synthetics.alerts.monitorStatus.removedLocation', {
-          defaultMessage: `Location has been removed from the monitor`,
-        });
+        recoveryStatus = i18n.translate(
+          'xpack.synthetics.alerts.monitorStatus.removedLocation.status',
+          {
+            defaultMessage: `has recovered`,
+          }
+        );
+        recoveryReason = i18n.translate(
+          'xpack.synthetics.alerts.monitorStatus.removedLocation.reason',
+          {
+            defaultMessage: `This location has been removed from the monitor`,
+          }
+        );
       }
     }
 
     if (state?.idWithLocation && upConfigs[state.idWithLocation]) {
-      isUp = Boolean(upConfigs[state.idWithLocation]) || false;
-      recoveryReason = i18n.translate('xpack.synthetics.alerts.monitorStatus.upCheck', {
-        defaultMessage: `Monitor has recovered with status Up`,
+      const { idWithLocation, configId, locationId, errorStartedAt } = state;
+      const upConfig = upConfigs[idWithLocation];
+      isUp = Boolean(upConfig) || false;
+      const upTimestamp = upConfig.ping['@timestamp'];
+      const checkedAt = moment(upTimestamp).format('HH:MM:SS on DD/MM/YYYY');
+      const duration = getErrorDuration(moment(errorStartedAt), moment(upTimestamp));
+      recoveryStatus = i18n.translate('xpack.synthetics.alerts.monitorStatus.upCheck.status', {
+        defaultMessage: `is now Up`,
       });
+      recoveryReason = errorStartedAt
+        ? i18n.translate('xpack.synthetics.alerts.monitorStatus.upCheck.reasonWithDuration', {
+            defaultMessage: `The monitor returned to an Up state at {checkedAt} lasting {duration}`,
```

Comment:
This is the content specified in the issue from Drew, but I'm not sure if it makes sense. It kinda sounds like the "up" check lasts the duration, when really it's the duration of the down state.

Reply:
Not entirely sure I get it right. Do we mean something like:
`The monitor has recovered at {checkedAt} after being Down for {duration}`?